### PR TITLE
Check buffer size when frameWidth * frameHeight bigger than allocated buffer size

### PR DESCRIPTION
**Pull request by JJJoonngg**
- issue nubmer #21149 report, some android 12 device, allocated bufferSize is too small for the retrieve frame, so change logic for debugging
- when this debugging successful in android studio environment, I will change logic about check bufferSize for the work well in any android environment (any sdk version including 31 - android12)

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Android pack,docs
android_pack_config=ndk-18-api-level-24.config.py
```

### CODE REVIEW

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

Please take a look on comments below.

File: modules/videoio/src/cap_android_mediandk.cpp [hunk 1]
```diff
@@ -97,7 +97,29 @@ class AndroidMediaNdkCapture : public IVideoCapture
...
+                        {
+                            LOGE("Buffer size is too small. Frame is ignored. Enable verbose logging to see actual values of parameters");
+                            return false;
+                        }
+                    }
+                    if (info.flags & AMEDIACODEC_BUFFER_FLAG_END_OF_STREAM)
+                    {
                         LOGV("output EOS");
                         sawOutputEOS = true;
                     }
```
[the remark this answers is not in the document]
Comment (JJJoonngg) [reply modules/videoio/src/cap_android_mediandk.cpp:125]: Yes, Thanks your review, I will fix it and repush it!

File: modules/videoio/src/cap_android_mediandk.cpp [hunk 2]
```diff
@@ -97,7 +97,29 @@ class AndroidMediaNdkCapture : public IVideoCapture
                     LOGV("buffer size: %zu", bufferSize);
                     LOGV("width (frame): %d", frameWidth);
                     LOGV("height (frame): %d", frameHeight);
-                    if (info.flags & AMEDIACODEC_BUFFER_FLAG_END_OF_STREAM) {
+                    if (frameWidth * frameHeight * 3 / 2 > bufferSize)
+                    {
+                        if (bufferSize == 3110400 && (frameWidth == 1920 && frameHeight == 1088) ||  frameWidth == 1088 && frameHeight == 1920)
```
[the remark this answers is not in the document]
Comment (JJJoonngg) [reply modules/videoio/src/cap_android_mediandk.cpp:102]: Yes! Thank you

[the remark this answers is not in the document]
Comment (alalek) [reply modules/videoio/src/cap_android_mediandk.cpp]: Thank you for update!

Please remove unnecessary whitespace: http://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/33151/steps/whitespace%20opencv/logs/stdio

Also please remove double space on this line:
```
else if(bufferSize == 3110400 &&  frameWidth == 1088 && frameHeight == 1920)
                                ^^
```

[the remark this answers is not in the document]
Comment (alalek) [reply modules/videoio/src/cap_android_mediandk.cpp]: We still have whitespace issue: http://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/33152/steps/whitespace%20opencv/logs/stdio

```
modules/videoio/src/cap_android_mediandk.cpp:108: trailing whitespace.
+                        {   
                          ^^^ 3 spaces here
```

**Review by alalek — APPROVED** [review]
LGTM :+1:

**Pull request merged**